src/bin/cli/mod.rs

```diff
@@ -35,6 +36,18 @@ pub enum Commands {
 
     #[command(about = "Generate Solidity interface files from Anchor IDL files")]
     Idl(IdlCommand),
+
+    #[command(about = "Create a new project with a ready example")]
+    New(New),
+}
+
+#[derive(Args)]
+pub struct New {
```

Comment:
I disagree, I don't see how this could not be very obvious that it's the struct for the `new` subcommand. Additionally it's confusing if the name diverges from the subcommand name

Reply:
It might make sense here, but `fn new(args: New)` in line 70 from `src/bin/solang.rs` deserves either another name or a doc comment. I suggested a new name here because of this awkward construct.

Reply:
Hmm yeah that's a good point, the `new` function should definitively be renamed.

Reply:
Should this be called `init`, just like `git init`?

Reply:
cargo allows `cargo init` and `cargo new`. `git new` does not exist

Reply:
But there is `cargo contract new`